internal/cluster/connection_manager.go

```diff
@@ -60,16 +60,16 @@ const serializationVersion = 1
 var ClientVersion = "1.0.0"
```

Comment:
I am pretty unfamiliar with the code base so I am asking here :smile: 

So, where do we translate private addresses (for cloud etc.) to public addresses while trying to connect to initial members (which should come as private addresses) or trying to reconnect to last known addresses?

Reply:
For the cloud the initial address list comes from the address provider: https://github.com/yuce/hazelcast-go-client/blob/hazelcast-cloud-discovery/internal/cloud/address_provider.go for anything else it is: https://github.com/yuce/hazelcast-go-client/blob/hazelcast-cloud-discovery/internal/cluster/address_provider.go

Address translation for cloud is provided by: https://github.com/yuce/hazelcast-go-client/blob/hazelcast-cloud-discovery/internal/cloud/address_translator.go and for others at: https://github.com/yuce/hazelcast-go-client/blob/hazelcast-cloud-discovery/internal/cluster/address_translator.go (when discovery is enabled, defaultPublicAddressTranslator  otherwise defaultAddressTranslator .

When members are added/removed, the following event handler runs: https://github.com/yuce/hazelcast-go-client/blob/ab6d150c89ab7b36e5b312adc9cb08a5200a3531/internal/cluster/cluster_service.go#L131 Eventually addMember runs: https://github.com/yuce/hazelcast-go-client/blob/ab6d150c89ab7b36e5b312adc9cb08a5200a3531/internal/cluster/cluster_service.go#L277 which translates member addresses.

Reply:
I see, so we always store the translated addresses in the member list and other places and use it, right? I couldn't find any problems with this approach, so I am marking this as resolved